### PR TITLE
tmt/utils: add support for changing logging for run method

### PR DESCRIPTION
**Pull request by thrix**
Can come handy if you want to display run command output under different log level.

Signed-off-by: Miroslav Vadkerti <mvadkert@redhat.com>

### CODE REVIEW

**Comment by thrix** [comment]
ha, need to fix the tests

File: tmt/steps/provision/localhost.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def _prepare_ansible(self, what):
         # Playbook paths should be relative to the metadata tree root
         playbook = os.path.join(self.step.plan.run.tree.root, what)
         # Run ansible against localhost, in verbose mode, enable --become
-        self.run(f'ansible-playbook -vb -c local -i localhost, {playbook}')
+        self.run(f'ansible-playbook -vb -c local -i localhost, {playbook}', log=self.verbose)
```
Comment (pvalena) [inline tmt/steps/provision/localhost.py]: Ah, I was gonna fix #48, but as you're touching it: We need to add `shell=True`. Without it we can't handle pipes, output redirection etc.

```
subprocess.Popen( . . . , shell=True)
```

File: tmt/utils.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ def debug(self, key, value=None, color=None, shift=1):
         if self.opt('debug'):
             echo(self._indent(key, value, color, shift))
 
-    def _run(self, command, cwd):
+    def _run(self, command, cwd, log):
```
Comment (pvalena) [inline tmt/utils.py]: Ah, I was gonna fix #48, but as you're touching it: We need to add `shell=True`. Without it we can't handle pipes, output redirection etc.

```
subprocess.Popen( . . . , shell=True)
```

Comment (pvalena) [reply tmt/utils.py]: Not sure whether it should be default, but for `execute` it has to be, as Vagrant's execute is, and there's no easy way to change it.

Comment (thrix) [reply tmt/utils.py]: ack, will add it

Comment (thrix) [reply tmt/utils.py]: @pvalena added, sorry for the delay!

**Comment by thrix** [comment]
tests fixed. @psss & @pvalena kindly asking for review

**Comment by thrix** [comment]
@psss feel free to close this if we will have a better solution as this :)

**Comment by pvalena** [comment]
LGTM. This `shell` capability is also needed for some call in `Prepare` AFAIR.

**Comment by psss** [comment]
As discussed with @thrix in person, I'm not sure this change is aligned with how the logging is currently implemented: It gives user control to say how much details they want to see while this patch gives the control to a specific method.

I would like to be able to both execute `run()` without producing any output (in the default case) and see detailed output when running with `--debug`. Hardcoding the log level in particular methods breaks this feature.

Currently it's possible to set the logging level for each test steps individually. If we need even more granularity I'd suggest to invent some systematic solution which would be accessible from the command line as well, such as detecting special environment variables for more in-depth expert debugging.

**Comment by psss** [comment]
As discussed in person we'll improve the error logging in a different way, see #82 for the current outline.

**Pull request closed without merging**